stdlib/public/core/With.swift

```diff
@@ -0,0 +1,37 @@
+//===--- With.swift -------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Implementation of the `with` function that is implicitly available on all values.
+@inlinable
+@_disfavoredOverload
+@backDeployed(before: SwiftStdlib 5.10)
```

Comment:
Is this the best way to handle availability here?

Reply:
The `stability-stdlib-abi-with-asserts` test is failing with:

```diff
 Func _prespecialize() is a new API without @available attribute
 Func _stdlib_isOSVersionAtLeastOrVariantVersionAtLeast(_:_:_:_:_:_:) is a new API without @available attribute
+Func _with(_:) is a new API without @available attribute
+Func _with_throws(_:) is a new API without @available attribute
 Protocol CodingKey has added inherited protocol Sendable
```

Does this need an `@available` attribute (e.g. `@available(before: SwiftStdlib 5.10)`? It seems like we wouldn't want this to depend on any specific standard library version if it can be back-deployed.